openshift/ocm-agent-operator
# OSD-9966 Deploy a NetworkPolicy to restrict ingress to OA

### What type of PR is this?
Feature

### What this PR does / why we need it?
OCM Agent Operator now deploys a `NetworkPolicy` to restrict ingress traffic to OCM Agent to just the `alertmanager` pods running in `openshift-monitoring`.

This prevents easy abuse of the agent by other resources running in the cluster.

As part of this change I took the opportunity to perform some small refactoring to tidy up our tests by moving the `testOcmAgent` to a test const that each test can use.

### Which Jira/Github issue(s) this PR fixes?

[OSD-9966](https://issues.redhat.com/browse/OSD-9966)

### How to test

If you are testing the PR via a built image run on-cluster, then note that there are additional RBAC rules required to allow `ocm-agent-operator` to manage NetworkPolicies.


If you are testing via running the operator locally and using kubeadmin creds, the RBAC changes are not needed for a successful test.

Either way, once the operator is running, you can verify it by:

```bash
# First check the networkpolicy was installed
$ oc get networkpolicy -n openshift-ocm-agent-operator
NAME                                POD-SELECTOR    AGE
ocm-agent-allow-only-alertmanager   app=ocm-agent   6m50s

# Then verify that Prometheus has no access
$ oc rsh -n openshift-monitoring prometheus-k8s-0
Defaulting container name to prometheus.
sh-4.4$ curl http://ocm-agent.openshift-ocm-agent-operator.svc.cluster.local:8081/livez
(will time out)

# Then verify that Alertmanager has access
$ oc rsh -n openshift-monitoring alertmanager-main-0
Defaulting container name to alertmanager.
sh-4.4$ curl http://ocm-agent.openshift-ocm-agent-operator.svc.cluster.local:8081/livez
{"Status":"ok"}
```

### Pre-checks (if applicable):
- [X] Ran `make generate` command locally to validate code changes
